### ISSUE TITLE
'On Display' filter not opening automatically on page load

### ISSUE THREAD

**Issue by jamieu** [TheScienceMuseum/collectionsonline#1165]
http://collection.sciencemuseum.org.uk/search/gallery/national-railway-museum,-warehouse

### Page state on load (current behaviour—incorrect)
<img width="1238" alt="screen shot 2018-08-06 at 12 41 34" src="https://user-images.githubusercontent.com/91365/43714767-18a5fcf0-9976-11e8-88b0-84bbf0ca95cb.png">

### Page state on load (expected behaviour—correct)
<img width="1214" alt="screen shot 2018-08-06 at 12 41 31" src="https://user-images.githubusercontent.com/91365/43714768-18b84388-9976-11e8-8838-7e4ed7c16e5f.png">

**Comment by iteles** [comment]
The filtering behaviour appears to be working, but the `On Display` sidebar area isn't toggled open or checked.

**Comment by SimonLab** [comment]
The url on the first issue is:
 http://collection.sciencemuseum.org.uk/search/gallery/national-railway-museum,-warehouse 

but I think it is mal formed (the comma shouldn't be used to separate museum and gallery parameter values in the url) and should be:
http://collection.sciencemuseum.org.uk/search/museum/national-railway-museum/gallery/warehouse

When running the first url we have also the following message log on the server:
> Gallery didn't match, ensure the exact gallery title was used

I've try to reproduce the original url (the one with the comma) on the application and I double check the code if there were any rules describing how to transform the url with comma to filters but couldn't spot anything on this.

The way we are building the filters from the query parameters is by using Joi validation.
For example `/search/museum/national-railway-museum/gallery/warehouse` will be converted to an object with `Joi.validate` and the `filterSchema` functions:
https://github.com/TheScienceMuseum/collectionsonline/blob/736a0f7c4dbd30100dc7bd71dc841833aaf73c05/routes/search.js#L34-L38

The `filterSchema` function will rename the parameter `museum` and `gallery` to `filter[museum]` and filter[gallery] to match the format of the [jsonapi](https://jsonapi.org/):
https://github.com/TheScienceMuseum/collectionsonline/blob/736a0f7c4dbd30100dc7bd71dc841833aaf73c05/schemas/filter.js#L82-L83

In our case (with the first url) the filter value for the gallery is `national-railway-museum, warehouse` (the dash is replaced by a space) but the filter value for the museum is `null`. The reason why we still have the similar result is because Elasitcsearch will still look for results containing the location `national-railway-museum, warehouse` which will match most of the objects in the national railway museum and in the gallery warehouse. This is done via the `location` property in Elasticsearch:
https://github.com/TheScienceMuseum/collectionsonline/blob/736a0f7c4dbd30100dc7bd71dc841833aaf73c05/lib/facets/create-filters.js#L141-L150

So to recap:
- the url should be `/search/museum/national-railway-museum/gallery/warehouse` and not `/search/gallery/national-railway-museum,-warehouse`
- the current url in the top issue returns the same result because Elasticsearch will try to find all the objects in the location `national-railway-museum, warehouse`
- the checkboxes are not selected because the filter museum is `null` and the filter for the gallery is `national-railway-museum, warehouse` which doesn't match any galleries.

@jamieu at the end I don't think there is anything to fix on this issue and I double checked the code to make sure we didn't miss anything. Let me know if all of this makes sense :+1:
